Adding blog post draft for toqito. (#550)
unitaryfund/unitary.fund: src/content/blog/2024_toqito.md

---
title: "Toqito: Quantum Information Science Impact through Open Source"
author: Vincent Russo and Purva Thakre
day: 19
month: 9
year: 2024
tags:
    - python
    - toqito
    - microgrant
    - qis
---

![](/images/toqito_logo.png)

In early 2020, the humble [initial commit](https://github.com/vprusso/toqito/commit/a2f2449c4b9de81becff41cb25f1bca3fa180e8b) of the toqito project ([repo](https://github.com/vprusso/toqito)|[docs](https://toqito.readthedocs.io/en/latest/index.html)) made its way onto GitHub.

> ![][/images/toqito_first_commit.png]

Reviewer: Commenting to resolve this later: 

None of the images except for the logo show up in the preview. 

https://unitary-fund-prel-git-add-toqito-blog-post-unitary-fund.vercel.app/posts/2024_toqito

> toqito was launched as a personal research-oriented software project to help me perform more effective research in quantum information. A  tool like this would have been exceedingly helpful to me as a PhD student as it would have allowed me to rapidly test ideas and accelerate the pace and progress of my research.

Reviewer: ```suggestion
toqito was launched as a personal research-oriented software project to help Vincent perform more effective research in quantum information. A  tool like this would have been exceedingly helpful to Vincent as a PhD student as it would have allowed him to rapidly test ideas and accelerate the pace and progress of his research.
```

> toqito was launched as a personal research-oriented software project to help Vincent perform more effective research in quantum information. A  tool like this would have been exceedingly helpful to Vincent as a PhD student as it would have allowed him to rapidly test ideas and accelerate the pace and progress of his research.

Reviewer 2: ```suggestion
toqito was launched as a personal research-oriented software project to help Vincent Russo perform more effective research in quantum information. A  tool like this would have been exceedingly helpful to Vincent as a PhD student as it would have allowed him to rapidly test ideas and accelerate the pace and progress of his research.
```

In the subsequent four years since its inception, toqito has seen significant contribution, development, and usage for different problems in quantum information.

![][/images/toqito_star_chart.png]

> toqito is an open-source library for studying various objects in quantum information, namely, states, channels, and measurements. toqito provides numerical tools to study problems about entanglement theory, nonlocal games, and other aspects of quantum information often associated with computer science.

Reviewer 3: You may want to mention that toqito is a _Python_ library, since that's what distinguishes it from, say, QETLAB. In fact, if toqito is inspired by QETLAB, you may also want to mention that.

Author: Good suggestion. Added in 68f1fb9

> In the early days of toqito, I showed some basic functionality to co-workers with a quantum information background. One of them suggested the Unitary Fund’s [microgrant program](https://unitary.fund/grants/). To my delight, toqito was accepted as a microgrant recipient after making a short video for the application. Not only was the monetary reward a nice boost of encouragement, but the associated support from the UnitaryFund team pushed me to reach out to other scientists and researchers who may benefit from toqito and potentially guide its development. 

Reviewer: ```suggestion
In the early days of toqito, Vincent showed some basic functionality to co-workers with a quantum information background. One of them suggested the Unitary Fund’s [microgrant program](https://unitary.fund/grants/). To his delight, toqito was accepted as a microgrant recipient after making a short video for the application. Not only was the monetary reward a nice boost of encouragement, but the associated support from the UnitaryFund team pushed him to reach out to other scientists and researchers who may benefit from toqito and potentially guide its development. 
```

> One of the opportunities suggested by the Unitary Fund to promote toqito included a talk to the  [New York Quantum Computing Meetup group](https://www.meetup.com/new-york-quantum-computing-meetup/) participants. I recorded the subsequent [first video on the UnitaryFund YouTube channel](https://www.youtube.com/watch?v=6R7qSszJwBI) that outlined the basic premise of toqito at that time. To promote toqito to a broader audience, a short [whitepaper](https://joss.theoj.org/papers/10.21105/joss.03082) was published in the Journal of Open Source Software.

Reviewer: ```suggestion
One of the opportunities suggested by the Unitary Fund to promote toqito included a talk to the  [New York Quantum Computing Meetup group](https://www.meetup.com/new-york-quantum-computing-meetup/) participants. Vincent recorded the subsequent [first video on the UnitaryFund YouTube channel](https://www.youtube.com/watch?v=6R7qSszJwBI) that outlined the basic premise of toqito at that time. To promote toqito to a broader audience, a short [whitepaper](https://joss.theoj.org/papers/10.21105/joss.03082) was published in the Journal of Open Source Software.
```

> Later that year, the first [UnitaryHACK 2021](https://unitary.fund/posts/unitaryhack2021/) event took place. This hackathon allows maintainers of quantum open-source (QOSS)  software repositories to participate by adding a set of bountied issues to improve the participating QOSS project. This hackathon also involved some of the earliest external contributors to toqito. 

Reviewer 3: uber-nit: extra white space after `(QOSS)`

Author: Eagle eye. Fixed in 68f1fb9

Toqito has since continued to be a participating project at subsequent UnitaryHACKs ([2022](https://unitary.fund/posts/2022unitaryhack/), [2023](https://unitary.fund/posts/2023_unitaryhack/), and [2024](https://unitaryhack.dev/)) and has gained more users, contributors, and essential features that have been invaluable to the broader community. 

> As toqito continued to gain usage and traction, one of the prior contributors of UnitaryHACK 2023 ([Purva Thakre](https://github.com/purva-thakre)) applied for and received a second [microgrant](https://unitary.fund/grants/) from UnitaryFund to continue supporting the project's development. These funds allowed dedicated time to delve into some of the more in-depth feature requests and continue to improve additional aspects of the project, such as thorough testing, documentation, CI/CD, and tutorials. 

Reviewer: ```suggestion
As toqito continued to gain usage and traction, another [UF microgrant](https://unitary.fund/grants/) to further develop the project was awarded a prior contributor of UnitaryHACK 2023 ([Purva Thakre](https://github.com/purva-thakre)). These funds allowed dedicated time to delve into some of the more in-depth feature requests and continue to improve additional aspects of the project, such as thorough testing, documentation, CI/CD, and tutorials. 
```

> In addition to being cited and used in peer-reviewed research papers on quantum information (a complete list can be found [here](https://github.com/vprusso/toqito?tab=readme-ov-file#references)), toqito has also been recognized elsewhere. Toqito was voted one of the [top quantum simulators for 2024 by QuantumInsider](https://thequantuminsider.com/2022/06/14/top-63-quantum-computer-simulators-for-2022/). Recently, [KaiCode](https://www.kaicode.org/2024.html) awarded toqito the first place prize for being the best project of 400+ projects judged on clean code, good project structure, etc.

Reviewer: I think this post is also missing the recently awarded microgrant.

> In addition to being cited and used in peer-reviewed research papers on quantum information (a complete list can be found [here](https://github.com/vprusso/toqito?tab=readme-ov-file#references)), toqito has also been recognized elsewhere. Toqito was voted one of the [top quantum simulators for 2024 by QuantumInsider](https://thequantuminsider.com/2022/06/14/top-63-quantum-computer-simulators-for-2022/). Recently, [KaiCode](https://www.kaicode.org/2024.html) awarded toqito the first place prize for being the best project of 400+ projects judged on clean code, good project structure, etc. Furthermore, as of September 2024, toqito is now an affiliated project of NumFOCUS.

Reviewer 2: This could be expanded, see main comment.
